### PR TITLE
feat(user): updating user query

### PR DESCRIPTION
**Pull request by timc-mozilla**
## Summary
* move isPremium to own query

## References 
* [IN-1198](https://getpocket.atlassian.net/browse/IN-1198)

## Implementation Details
* moved isPremium out of fetch saves and into its own query

## Test Steps
* as non-premium user, I should see all upsells
* as premium user, I should not see any upsells

## PR Checklist:
- [ ] Added Unit / UI tests
- [ ] Self Review (review, clean up, documentation, run tests)
- [ ] Basic Self QA

## Screenshots

[IN-1198]: https://getpocket.atlassian.net/browse/IN-1198?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: PocketKit/Sources/Sync/User/UserService.swift [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+            user.setPremiumStatus(remoteIsPremium)
+            user.setUserName(remoteUserName)
+
+            if let remoteDisplayName = remote.username {
+                user.setDisplayName(remoteDisplayName)
+            } else {
+                Log.capture(message: "No display name data")
+            }
+        }
+        try space.save()
```
Comment (bassrock) [inline PocketKit/Sources/Sync/User/UserService.swift:51]: Do we need to save the space? Is the user in coredata?

**Review by bassrock — COMMENTED** [review]
Just have one comment then i can approve

File: PocketKit/Sources/Sync/User/UserService.swift (hunk 1 above)
Comment (timc-mozilla) [reply PocketKit/Sources/Sync/User/UserService.swift:51]: yup, youre right. it was a part of a previous solution i had for it. ill remove it and run tests again.

**Review by bassrock — APPROVED** [review]

**Comment by timc-mozilla** [comment]
all tests pass locally 
image

**Pull request merged**